variables.tf

```diff
@@ -3,6 +3,11 @@ variable "naming_rules" {
   type        = string
 }
 
+variable "dnsserver" {
+  description = "custom dns server" 
+  type        = list(string)
+  default     = []

```

Comment:
The default value for this should be:  null

If the dns_servers attribute is not a required input for the azurerm_virtual_network resource, so if no dns_servers are specified, we can just use the Azure defaults.  The way to do that is to set the default value to null (no quotes).

Reply:
added as default = null